css_coding_styleguide.md

```diff
@@ -0,0 +1,755 @@
+
+```scss
+// Good
+selector1,
+selector2,
+selector3 {
+}
+
+// Good
+h1, h2 {
+}
+
+// Bad 
+selector1, selector2 {
+}
+```
+
+### Property-Value Pairs
+- Put each pair on its own line.
+- Indent each pair one level.
+- End in a semicolon.
+
+```scss
+selector {
+  name: value;
+  name: value;
+}
+```
+
+
+- Do not use shorthand declarations unless you need to explicitly set all the available values.
+
+```scss
+// Bad
+margin: inherit 3em;
+
+// Good
+margin-bottom: 3em;
+margin-top: 3em;
+
+margin: 3em 4em 2em 1em;
+```
+
+- Single-quote URLs and string values.
+
+```scss
+  background-image: url('/images/kittens.jpg');
+  font-family: 'Helvetica', sans-serif;
+  font-family: 'Lucida Grande', 'Helvetica', sans-serif;
+```
+
+- Wrap top-level numeric calculations in parentheses.
+
+```scss
+// Good
+.component {
+  width: (100% / 3);
+}
+
+// Bad
+.component {
+  width: 100% / 3;
+}
+```
+
+- Avoid arbitrary numbers that are repeated, or linked, or dependent on other parts of the code, (aka “magic numbers”).
+
+```scss
+// Bad
+.component {
+  top: 0.327em;
+}
+
+// Better
+/**
+ * 1. Magic number. This value is the lowest I could find to align the top of
+ * `.foo` with its parent. Ideally, we should fix it properly.
+ */
+.component {
+  top: 0.327em;
+}
+
+// Good
+$align_top: 100%;
+.component {
+  top: $align_top;
+}
+```
+
+### Sorting
+* Follow ordering:
+	1. variables
+	2. @extend directives
+	3. @include directives
+	4. properties
+- Within properties, you may use alphabetical order or type order—just pick one and keep the whole project consistent.
+- Put a new line before nested selectors.
+- Put mixin calls with @content after nested selectors.
+- Properties should be alphabetical, not grouped by type or length.
+
+```scss
+// Good
+.module {
+  $amount = 3;
+  @extend .component;
+  @include sizing($amount);
+  margin-top: $amount * 1em;
+  text-align: center;
+
+  .module__ele {
+    color: #fff932;
+  }
+
+  @include media($sm) {
+    margin-top: $amount +10em;
+  }
+}
+```
+
+<a name="format_notes"></a>
+
+#### Notes
+##### How to set text editors to 80 chars
+- Sublime: Add a `rulers` setting with 80 as the value.
+	- ```"rulers": [80]```
+- Atom: Set the `preferredLineLength` setting to 80.
+	- `preferredLineLength: 80`
+- Vim: Set two options in your .vimrc to wrap lines at 80 characters.
+	- ```set formatoptions+=w```
+	- ```set tw=80```
+
+
+## Units
+### Measurements
+- Use **rem** units for font sizes with a px fallback.
+This can be done with the following mixin:
+
+```scss
+@mixin font-size($sizeValue: 1.6) {
+  font-size: ($sizeValue * 10) + px;
+  font-size: $sizeValue + rem;
+}
+```
+
+- Set the html font size to 10px to ensure .1 rem unit equals 1px.
+
+```scss
+html {
+  font-size: 10px;
+}
+```
+
+- Use **em** units for positioning.
+- Use **percentages** when elements have to change based on screen size.
+- Use **px** units for when a measurement shouldn't change based on user set font size or browser zooming.
```

Comment:
maybe we can add an example / context... 1-5px for detail work is fine (e.g. borders, lines, etc)